pandas/tests/plotting/frame/test_frame.py

```diff
@@ -1070,41 +1070,66 @@ def test_boxplot_series_positions(self, hist_df):
         tm.assert_numpy_array_equal(ax.xaxis.get_ticklocs(), positions)
         assert len(ax.lines) == 7 * len(numeric_cols)
 
+    @pytest.mark.filterwarnings("ignore:set_ticklabels:UserWarning")
+    @pytest.mark.xfail(
+        Version(mpl.__version__) >= Version("3.10"),
+        reason="Fails starting with matplotlib 3.10",
+    )
     def test_boxplot_vertical(self, hist_df):
         df = hist_df
         numeric_cols = df._get_numeric_data().columns
         labels = [pprint_thing(c) for c in numeric_cols]
 
         # if horizontal, yticklabels are rotated
-        ax = df.plot.box(rot=50, fontsize=8, vert=False)
+        kwargs = (
+            {"vert": False}
+            if Version(mpl.__version__) < Version("3.10")
+            else {"orientation": "horizontal"}
+        )
+        ax = df.plot.box(rot=50, fontsize=8, **kwargs)
```

Comment:
Just to understand the change: even though using the new keyword for mpl >= 3.10, it is still failing somehow? (given you added the xfail)

Reply:
Yeah that's correct. It appears that changing this to use the equivalent, new matplotlib 3.10 keyword causes some tests to fail. I was unable to determine if the failure was on our side or the matplotlib side